### DISCUSSION TITLE
how to write msgpack to file

### DISCUSSION THREAD

**Discussion by jiapengwen** [nlohmann/json#3182] [Q&A] [answered]
```c++
// serialize to MessagePack
std::vector<std::uint8_t> v_msgpack = json::to_msgpack(j);

//for json
ofstream outfile(file, ios::out | ios::trunc);
outfile << j.dump();
```

json support api write std::vector\<std::uint8_t\> to file? 

also read from file, then from_msgpack to json

**Comment by nlohmann** [comment, marked as answer]
This may help: https://stackoverflow.com/questions/14089266/how-to-correctly-write-vector-to-binary-file-in-c

**Reply by jiapengwen** [reply]
ok, maybe json could support a api to solve this problem

**Reply by nlohmann** [reply]
There is no need for this, as C++ already has all the required APIs to write binary files.
